R/projection.R

```diff
@@ -76,14 +100,38 @@ svd_project_with_ops <- function(scaling, ops) {
 #'
 #' @param scaling dso$st$scaling object containing sinkhorn scaling result
 #' @param dims how many dimensions we want to get
+#' @param ops = SVD projection operations. Default is NULL and calculated on the fly for backward compatability.
 #' @return proj object
 #' @export
-svd_project <- function(scaling, dims) {
-  ops <- calc_svd_ops(scaling$V_row, dims)
-  proj <- svd_project_with_ops(scaling, ops)
+svd_project <- function(scaling, dims, ops = NULL) {
+  if (is.null(ops)) ops <- calc_svd_ops(scaling$V_row, max_dim = max(dims))
+  proj <- svd_project_with_ops(scaling, ops, dims = dims)
   return(proj)
 }
 
+#' Prepare projection object
+#'
+#' Entry point to produce dso$st$proj object, containing all information about projection as well as projected points
+#'
+#' @param dso DualSimplexSolver object.
+#' @param dims a integer vector indicateing the dimensions we want to get.
+#' @param ops SVD projection operations. Default is NULL and calculated on the fly for backward compatability.
+#' @return proj object
+#' @export
+efficient_svd_project <- function(dso, dims, ops = NULL) {
```

Comment:
Methods from this R file should not work with original dso object. This creates situation when lower level entities modify higher level entities, which should not be done